web/docs/auth/migrate-from-11.md

```diff
@@ -0,0 +1,349 @@
+
+Expect some downtime when migrating a deployed app. To be more precise: the auth system won't work 100% while the migration is in progress.
+
+As soon as you deploy the new auth system, new users will be able to sign up, but existing users won't be able to log in until you run the migration script on the production database.
+
+You should probably back up your database before migrating a deployed app. Testing the changes locally or on a staging environment first is also a good idea.
+:::
+
+### 1. Migrate to the New Auth System
+
+If you have an existing app with authentication set up, you can follow these steps to migrate to the new auth system:
+
+1. Upgrade Wasp to the latest 0.12 version.
+1. Bump the version to `^0.12.0` in `main.wasp`.
+1. Create the new auth tables in the database by running:
+
+  ```bash
+  wasp db migrate-dev
+  ```
+
+  You should see the new `Auth`, `AuthIdentity` and `Session` tables in your database.
+
+1. Create a data migration script in `src/server/migrateToNewAuth.ts`
+
+    - Below you can find [examples of migration scripts](#example-migration-scripts) for each of the auth methods. They should be fine to use as-is, but you can also modify them to your needs.
+
+1. Add the migration script(s) to the `db.seeds` config:
+  ```wasp title="main.wasp"
+  app myApp {
+    wasp: {
+      version: "^0.12.0"
+    },
+    // ...
+    db: {
+      seeds: [
+        import { migrateEmailAuth } from "@server/migrateToNewAuth.js",
+        import { migrateGoogleAuth } from "@server/migrateToNewAuth.js",
+      ]
+    },
+  }
+  ```
+1. Run the migration script(s) by running:
+  ```bash
+  wasp db seed
+  ```
+  If you added multiple migration scripts, you can pick which one to run by selecting it from the list.
+
+1. Verify that the auth still works locally by logging in with each of the auth methods.
+1. Update your JS code to use the new auth helper functions for getting the `email` or `username` of the currently logged-in user. Read more about the helpers in the [Auth Entities](./entities#accessing-the-auth-fields) section.
+
+  The helpers you'll probably use are the `getEmail` and `getUsername` helpers.
+
+  :::info Migrating a deployed app
+
+    After you have run the migration script locally, updated your JS code and saw that it's okay, _deploy the app_ with the changes you have made so far.
+
+    Then you'll need to run the migration script on the production database with `wasp db seed` command.
+
+    **Running the migration script on the production database**
+
+    We wrote instructions on how to do it for **Fly.io** deployments here: https://github.com/wasp-lang/wasp/issues/1464 
+
+    The instructions should be similar for other deployment providers: setting up some sort of an SSH tunnel from your local machine to the production database and running the migration script locally with `DATABASE_URL` pointing to the production database.
+  :::
```

Comment:
```suggestion
  :::info Migrating a deployed app
  
    After successfully performing migration locally so far, and verifying the migrated app works as expected, it is time to also migrate our deployed app (if you have one).
    
    Before migrating your production (deployed) app, we advise you create a copy of your production database in case something goes wrong. Also, besides testing it in development, it might also be advisable to test the migration in the staging environment (if you have one).
    
    Once you are ready, we will perform the production migration in 2 steps:
    1. Deploying the new code to production (client and server).
    2. Migrating the production database.
    
    Between these two steps, so after deploying the new code to production and before you migrate the production database, your app will not be working completely: new users will be able to sign up, but existing users won't be able to log in, and already logged in users will be logged out. Once you do the second step, migrating the production database, it will all be back to normal.
    Therefore, you will likely want to keep the time between the two steps as short as you can. It is advisable to make sure you know exactly what each step entails before doing them for real and also giving them a dry run (or trying it in staging), to eliminate any surprises.
    
    First step: deploy the new code (client and server), either via `wasp deploy` or manually. Check our [Deployment docs](../advanced/deployment/overview.md) for more details. 

    Second step: run the migration script on the production database with `wasp db seed` command. We wrote instructions on how to do it for **Fly.io** deployments here: https://github.com/wasp-lang/wasp/issues/1464 . The instructions should be similar for other deployment providers: setting up some sort of an SSH tunnel from your local machine to the production database and running the migration script locally with `DATABASE_URL` pointing to the production database.
    
    This should be it for the moment! At this point, your deployed app should be working normally.
  :::
```

Reply:
We should also provide a link to Discord at some place and tell them to ask us for help if not sure about something.

Reply:
> giving them a dry run

How would they do a dry run of the migration steps? I've removed that bit, but also modified the text above to feel more natural to me.

Reply:
I would try connecting with an SSH tunnel first, without doing anything else, just to make sure that works. But you are right I left it kind of undefined, because I am not sure how much can they "dry run". Ok!

Reply:
I checked the new text you applied, looking good.